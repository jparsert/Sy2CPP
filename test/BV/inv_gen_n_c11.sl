; default grammar added
(set-logic BV)
(synth-fun inv((i (_ BitVec 32))(l (_ BitVec 32)))
 Bool
(( NTbool Bool)(NTnonbool (_ BitVec 32)))
((NTbool Bool((and NTbool NTbool)(or NTbool NTbool)(not NTbool)(= NTnonbool NTnonbool)(bvuge NTnonbool NTnonbool)(bvugt NTnonbool NTnonbool)))
(NTnonbool (_ BitVec 32)(i l (_ bv0 32) (_ bv1 32) (_ bv2 32) (_ bv3 32)  (bvadd NTnonbool NTnonbool)(bvsub NTnonbool NTnonbool)(bvshl NTnonbool NTnonbool)(bvlshr NTnonbool NTnonbool)(bvurem NTnonbool NTnonbool)(ite NTbool NTnonbool NTnonbool)))

)
)

(declare-var i (_ BitVec 32))
(declare-var i1 (_ BitVec 32))
(declare-var l (_ BitVec 32))
(declare-var l1 (_ BitVec 32))
(declare-var l2 (_ BitVec 32))
( constraint  (=> (= l (_ bv0 32)) (inv i l )) ) 
( constraint  (=> (and (and (and (and (inv i l ) (=> (= l (_ bv4 32)) (= l1 (_ bv0 32)))) (=> (not (= l (_ bv4 32))) (= l1 l))) (not (or (bvult l1 (_ bv0 32)) (bvuge l1 (_ bv5 32))))) (= l2 (bvadd l1 (_ bv1 32)))) (inv i l2 )) ) 
( constraint  (=> (and (and (and (inv i l ) (=> (= l (_ bv4 32)) (= l1 (_ bv0 32)))) (=> (not (= l (_ bv4 32))) (= l1 l))) (or (bvult l1 (_ bv0 32)) (bvuge l1 (_ bv5 32)))) false) ) 
(check-synth)
