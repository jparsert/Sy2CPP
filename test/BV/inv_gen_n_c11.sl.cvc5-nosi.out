(
(define-fun inv ((i (_ BitVec 32)) (l (_ BitVec 32))) Bool (bvugt (bvadd #b00000000000000000000000000000010 #b00000000000000000000000000000011) l))
)
1.95user 0.00system 0:01.95elapsed 100%CPU (0avgtext+0avgdata 32940maxresident)k
0inputs+8outputs (0major+4148minor)pagefaults 0swaps
